service/src/main.rs

```diff
@@ -575,24 +576,35 @@ fn execute_update_market<E: TeeracleApi>(node_api: &Api<sr25519::Pair, WsRpcClie
 			Ok(r) => r,
 		};
 
-	let mut hex_encoded_extrinsic = hex::encode(updated_extrinsic);
-	hex_encoded_extrinsic.insert_str(0, "0x");
-
-	// Send the extrinsic to the parentchain and wait for InBlock confirmation.
-	println!("[>] Update the exchange rate (send the extrinsic)");
-	let extrinsic_hash = match node_api.send_extrinsic(hex_encoded_extrinsic, XtStatus::InBlock) {
+	let extrinsics: Vec<OpaqueExtrinsic> = match Decode::decode(&mut updated_extrinsic.as_slice()) {
+		Ok(calls) => calls,
 		Err(e) => {
 			error!("{:?}: ", e);
-			set_extrinsics_inclusion_success(false);
 			return
 		},
-		Ok(r) => {
-			set_extrinsics_inclusion_success(true);
-			r
-		},
 	};
 
-	println!("[<] Extrinsic got included into a block. Hash: {:?}\n", extrinsic_hash);
+	// Send the extrinsics to the parentchain and wait for InBlock confirmation.
+	for call in extrinsics.into_iter() {
+		let mut hex_encoded_extrinsic = hex::encode(call.encode());
+		hex_encoded_extrinsic.insert_str(0, "0x");
+
+		println!("[>] Update the exchange rate (send the extrinsic)");
+		let extrinsic_hash = match node_api.send_extrinsic(hex_encoded_extrinsic, XtStatus::InBlock)
```

Comment:
Is it not a problem to wait until InBlock for every extrinsic that needs to be sent? (InBlock takes  up to 12 secs for every extrinsic).

Reply:
That's a good point - is there an alternative? Can we run this in a non-blocking fashion?

Reply:
Couldn't we run in asynchronously, and update the metrics upon an `InBlock` event? Or we could also run it within multiple threads, as we are on the untrusted side. What do you think?

Reply:
Ah! We're in the untrusted side here, I didn't realize that at first. Then yes, we can use asynchronous patterns. We can use `tokio` to spawn asynchronous tasks here. Pass a `tokio` handle into this method, then in this iteration here, call: 
```
tokio_handle.spawn(move { 
		let mut hex_encoded_extrinsic = hex::encode(call.encode());
		hex_encoded_extrinsic.insert_str(0, "0x");

		println!("[>] Update the exchange rate (send the extrinsic)");
		let extrinsic_hash = match node_api.send_extrinsic(hex_encoded_extrinsic, XtStatus::InBlock) {
                          ...
                };

                println!("[<] Extrinsic got included into a block. Hash: {:?}\n", extrinsic_hash);
 } );
```

Just let me know if you need help with that..